src/services/PaymentService.ts

```diff
@@ -188,6 +188,9 @@ export class PaymentService {
       discounts: couponCode ? [{ coupon: couponCode }] : undefined,
       allow_promotion_codes: couponCode ? undefined : true,
       billing_address_collection: 'required',
+      subscription_data: {
+        trial_period_days: trial ? 30 : undefined,
```

Comment:
Add a trial or do not add it, but do not add a default trial, that would add trials to all subscriptions